:sparkles: feat: 모든 키워드 조회 (R) 구현

backswan0 opened the pull request:
## #️⃣ Kan Board Number

7, 21

## 📝 요약

사용자가 알림으로 등록하고 싶은 키워드를 모두 조회하는 기능 구현 - Postman 테스트 완료 

## 🛠️ PR 유형

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💬 논의사항 or 질문

40줄 밖에 없지만 사용자 채용 키워드 알림 조회 기능과 별도로 풀 리퀘스트 했습니다:) 뭔가 놓친 듯한 느낌이 있는데, 혼자 봐서는 파악이 안 되어서 놓친 부분이 있거나 더 좋게 개선할 부분이 있다면 피드백 적극적으로 부탁드리겠습니다:)

## ✅ PR Checklist

PR이 다음 요구 사항을 충족했는지 확인하기!

- [x] 커밋 메시지 컨벤션 준수
- [x] 변경 사항 테스트 여부 체크(버그 수정/기능 테스트)

## 💥 리뷰어 지목!!!!!

<!--- 리뷰어를 2명 이상 지목해주세요. -->
- [x] 채영
- [ ] 지현
- [x] 리은
- [x] 승찬
- [ ] 주양

backswan0 committed:
:sparkles: feat: 모든 키워드 조회 (R) 구현

- 사용자가 알림으로 등록할 수 있는 키워드 목록을 조회하는 API 구현

TaehoHong commented on src/main/java/com/project/cheerha/domain/keyword/dto/response/ReadKeywordResponseDto.java:
```diff
@@ -0,0 +1,10 @@
+
+import java.util.List;
+
+public record ReadKeywordResponseDto(List<String> allKeywordNameList) {
+
+    public static ReadKeywordResponseDto toDto(List<String> allKeywordNameList) {
+        return new ReadKeywordResponseDto(allKeywordNameList);
+    }
+}
```
Keyword Id는 반환 안해도 되나요?? 
Keyword Id를 사용하는 다른 API는 없는 건가요?

roqkfchqh replied:
id를 반환해줘야 add가 가능합니다

backswan0 replied:
네네, 지금 피드백 받아서 수정하겠습니다:)

backswan0 committed:
:sparkles: feat: 사용자 알림 채용 키워드 조회 (R) 구현

- 사용자가 본인이 어떤 키워드를 알림으로 등록했는지 조회하는 기능 구현

- [할 일] id 함께 반환하기

backswan0 committed:
:recycle: refactor: 모든 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

- 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

backswan0 committed:
:recycle: refactor: 사용자 알림 채용 키워드 조회 (R) 리팩토링

- 사용자가 알림으로 등록한 키워드 조회 시 키워드의 아이디와 이름을 함께 반환하도록 리팩토링

- [할 일] 머지 이후 사용자 알림 채용 키워드 등록 (C) 또한 위와 같은 로직으로 리팩토링
